### PR TITLE
tests: gather /var/run/cilium, /var/lib/cilium as part of Jenkins builds

### PR DESCRIPTION
**Pull request by ianvernon**
Gather contents of /var/run/cilium, /var/lib/cilium for each test that is ran in tests/ directory, as well as cilium logs, and package up and archive for each build job.
    
Signed-off by: Ian Vernon <ian@covalent.io>

Fixes: #1016

### CODE REVIEW

**Comment by ianvernon** [comment]
Logs / contents of /var/run/cilium, /var/lib/cilium are packaged up now in a tar.gz as part of the build. [Example](https://jenkins.cilium.io/job/cilium/job/cilium/job/PR-1017/33/).

**Comment by aanm** [comment]
Small thing can you remove one of the timestamps from the log?

**Review by tgraf — APPROVED** [review]

**Review by aanm — CHANGES_REQUESTED** [review]
Ping me once the duplicate timestamps are removed

**Review by aalemayhu — CHANGES_REQUESTED** [review]
Can you change `/home/vagrant/go/` to `$GOPATH`? This will make the script useable outside of vagrant environments.

File: tests/helpers.bash [hunk 1]
```diff
@@ -133,5 +127,20 @@ function wait_for_running_pod {
     while [[ "$(kubectl get pods | grep ${pod} | grep Running -c)" -ne "1" ]] ; do
         micro_sleep
     done
-    set -x
+}
+
+function gather_files {
+    TEST_NAME=$1
+    CILIUM_DIR="/home/vagrant/go/src/github.com/cilium/cilium/tests/cilium-files/$1"
```
Comment (aalemayhu) [inline tests/helpers.bash]: `TEST_NAME` is unused.

File: Jenkinsfile [hunk 2]
```diff
@@ -14,13 +14,15 @@ pipeline {
...
+                        "K8s Tests": { sh './tests/k8s/start' }
                     )
 	        }
         }
     }
     post {
         always {
+            sh './tests/copy_files'
+            archiveArtifacts "cilium-files.tar.gz"
             sh 'vagrant destroy -f'
```
Comment (aalemayhu) [inline Jenkinsfile:26]: Nice, this makes the artefact available for download from Jenkins

https://jenkins.cilium.io/job/cilium/job/cilium/job/PR-1017/35/artifact/cilium-files.tar.gz

**Comment by ianvernon** [comment]
@aanm given that the formatting of the logs was removed by another commit which is unrelated to the goal of this ticket, 41786b1ef , I'd prefer to file another ticket and deal with it separately. I'll happily take on that ticket as well. Is that OK with you?

**Review by aalemayhu — APPROVED** [review]
Other than the newlines in `run-tests`. This looks great!

**Comment by aanm** [comment]
>@aanm given that the formatting of the logs was removed by another commit which is unrelated to the goal of this ticket, 41786b1 , I'd prefer to file another ticket and deal with it separately. I'll happily take on that ticket as well. Is that OK with you?

Sounds good!

**Review by aanm — APPROVED** [review]

**Comment by ianvernon** [comment]
Logging issues will be taken care of separately: https://github.com/cilium/cilium/issues/1029

**Pull request merged**